pandas/core/dtypes/cast.py

```diff
@@ -357,12 +357,15 @@ def maybe_cast_result_dtype(dtype: DtypeObj, how: str) -> DtypeObj:
         The desired dtype of the result.
     """
     from pandas.core.arrays.boolean import BooleanDtype
+    from pandas.core.arrays.floating import Float64Dtype
     from pandas.core.arrays.integer import Int64Dtype
 
     if how in ["add", "cumsum", "sum"] and (dtype == np.dtype(bool)):
         return np.dtype(np.int64)
     elif how in ["add", "cumsum", "sum"] and isinstance(dtype, BooleanDtype):
```

Comment:
```suggestion
    elif how in ["add", "cumsum", "sum", "prod"] and isinstance(dtype, BooleanDtype):
```

prod as well?

Reply:
i was kind of hoping you'd take the lead in fleshing these out with appropriate testing

Reply:
I added some tests to start with.

Reply:
thanks